api/src/collection.py

```diff
@@ -1,36 +1,179 @@
+        }
+        return collection_formatted
+
+    def _create_zarr_template(self, dataset: ZarrDataset, store_path: str) -> dict:
+        template = self.get_template(dataset)
+        template["assets"] = {
+            "zarr": {
+                "href": store_path,
+                "title": f"{dataset.collection} Zarr root",
+                "description": "",
+                "roles": ["data", "zarr"],
+                "type": "application/vnd+zarr",
+                "xarray:open_kwargs": {
+                    "engine": "zarr",
+                    "chunks": {},
+                    **dataset.xarray_kwargs,
+                },
+            }
+        }
+        return template
+
+    def create_zarr_collection(self, dataset: ZarrDataset) -> dict:
+        """
+        Creates a zarr stac collection based off of the user input
+        """
+        s3_creds = get_s3_credentials()
+        discovery = dataset.discovery_items[0]
+        store_path = f"s3://{discovery.bucket}/{discovery.prefix}{discovery.zarr_store}"
+        template = self._create_zarr_template(dataset, store_path)
+        store = fsspec.get_mapper(store_path, client_kwargs=s3_creds)
+        ds = xr.open_zarr(
+            store, consolidated=bool(dataset.xarray_kwargs.get("consolidated"))
+        )
+
+        collection = xstac.xarray_to_stac(
+            ds,
+            template,
+            temporal_dimension=dataset.temporal_dimension or "time",
+            x_dimension=dataset.x_dimension or "lon",
+            y_dimension=dataset.y_dimension or "lat",
+            reference_system=dataset.reference_system or 4326,
+        )
+        return collection.to_dict()
+
+    def create_cog_collection(self, dataset: COGDataset) -> dict:
+        collection_stac = self.get_template(dataset)
+        collection_stac["extent"] = SpatioTemporalExtent.parse_obj(
+            {
+                "spatial": {
+                    "bbox": [
+                        list(dataset.spatial_extent.dict(exclude_unset=True).values())
+                    ]
+                },
+                "temporal": {
+                    "interval": [
+                        # most of our data uses the Z suffix for UTC - isoformat() doesn't
+                        [
+                            x.isoformat().replace("+00:00", "Z")
+                            for x in list(
+                                dataset.temporal_extent.dict(
+                                    exclude_unset=True
+                                ).values()
+                            )
+                        ]
+                    ]
+                },
+            }
+        )
+        collection_stac["item_assets"] = {
+            "cog_default": {
+                "type": "image/tiff; application=geotiff; profile=cloud-optimized",
+                "roles": ["data", "layer"],
+                "title": "Default COG Layer",
+                "description": "Cloud optimized default layer to display on map",
```

Comment:
![image](https://user-images.githubusercontent.com/3404817/220577859-3ff7e390-9fb0-4994-8c5d-02dad6eadbba.png)

Maybe we can find a better asset title? I think the concept of "layers" is not meaningful if you do not plan to add the data to a map (or load it in classical GIS software) but just read it into a process. 

A short title descriptive of the contents would be great (e.g. "NDVI" if the COG contained that). But we want to automate this. So if this needs to be static, I would suggest the following:

```suggestion
                "title": "COG",
                "description": "Raster data in Cloud Optimized GeoTIFF (COG) format",
```

Reply:
`cog_default` assets were created with the dashboard team to describe specific subset of assets that were intended for the dashboard (for a collection with item_assets B1, B2, B3, and cog_default it would be clearer). I think the current key and title are correct to advertise when the collection contains this specific kind of asset that is intended to be used to display in the dashboard by default. 

However, as we expand our holdings it will not be safe to assume that all collections will always have simple single band default_cogs. _In a future PR_ I think we should refactor so that the base class for a Collection is a pystac Collection and then potentially support a custom optional COG|Dashboard?|Dataset defined with the default pystac Collection as a base class.

Reply:
Ok, thanks for the context, @anayeaye.